packages/opencensus-core/src/stats/metric-producer.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {MetricProducer} from '../metrics/export/metric-producer';
+import {Metric} from '../metrics/export/types';
+
+import {Stats} from './stats';
+
+/**
+ * A MetricProducer producer that can be registered for exporting using
+ * MetricProducerManager.
+ */
+export class MetricProducerForStats extends MetricProducer {
+  private statsManager: Stats;
+
+  /**
+   * @param statsManager The stats
+   */
+  constructor(statsManager: Stats) {
+    super();
+    this.statsManager = statsManager;
+  }
+
+  /**
+   * Gets a collection of produced Metric`s to be exported.
+   * @returns {Metric[]} List of metrics
+   */
+  getMetrics(): Metric[] {
```

Comment:
I know you have covered ```getMetrics()``` method in another class. Could you pls add tests for this class using ```statsManager```.

Reply:
@mayurkale22 This functionality is not in scope of [#213](https://github.com/census-instrumentation/opencensus-node/issues/213) task

Reply:
Added base test, detailed in test-view.ts